source/util/StringUtils.cpp

```diff
@@ -60,6 +61,31 @@ namespace Aws
                     }
                     return output.str();
                 }
+
+                string addString(Aws::Crt::String first, Aws::Crt::String second)
+                {
+                    string jsonTemplate = R"("%s": "%s")";
+                    return FormatMessage(jsonTemplate.c_str(), first.c_str(), second.c_str());
+                }
+
+                string MapToJsonString(Crt::Optional<Crt::Map<Aws::Crt::String, Aws::Crt::String>> map)
+                {
+                    std::map<Aws::Crt::String, Aws::Crt::String>::iterator it;
+                    string result = "";
+                    unsigned int count = 0;
+                    for (it = map->begin(); it != map->end(); it++)
+                    {
+                        count++;
+                        result.erase(std::find(result.begin(), result.end(), '\0'), result.end());
```

Comment:
why do you need to run this erase on each iteration of the loop?

Reply:
I need to remove the null terminator character '\0'. Without doing that the next lines or string will not be written in the file. 
Example: For 
{a:b,
c:d}. Only a:b was written in the file.

Reply:
Can \0 exist in a key-pair

Reply:
Also shouldn't \0 just be removed at the end of the entire operation?

Reply:
Updated code to remove all '\0' in the end.